btcpy/structs/hd.py

```diff
@@ -20,33 +20,33 @@
 
 from ..lib.types import HexSerializable
 from ..lib.parsing import Stream, Parser
-from ..setup import is_mainnet
+from ..setup import is_mainnet, net_name
 from .crypto import PrivateKey, PublicKey
 
 
 class ExtendedKey(HexSerializable, metaclass=ABCMeta):
-    
-    master_parent_fingerprint = bytearray([0]*4)
+    master_parent_fingerprint = bytearray([0] * 4)
     first_hardened_index = 1 << 31
     curve_order = SECP256k1.order
+    networks = {
+        'x': 'mainnet',
+        't': 'testnet',
+    }
     
     @classmethod
     def master(cls, key, chaincode):
         return cls(key, chaincode, 0, cls.master_parent_fingerprint, 0, hardened=True)
     
     @classmethod
-    def decode(cls, string, check_network=True):
-        if string[0] == 'x':
-            mainnet = True
-        elif string[0] == 't':
-            mainnet = False
+    def decode(cls, string, check_network=False):
```

Comment:
Changing the default value of `check_network` is actually quite dangerous. This library should always default to a safe behaviour unless explicitly specified otherwise.

Reply:
It is outdated please check [next commit](https://github.com/chainside/btcpy/pull/17/commits/36d270f4a7ed090aca60caf06731d87dd5bf534c).

Reply:
Yeah, I saw that, just wanted to point it out to raise awareness on this kind of seemingly innocuous changes that can actually be dangerous.